### PR TITLE
Add section on architecture

### PR DESCRIPTION
**Pull request by wooorm**
/cc @ggoodman

See [new section rendered](https://github.com/wooorm/xdm/blob/9082c0ec5906b546d633a4a12cba882e779530f7/readme.md#architecture)

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1737,6 +1738,66 @@ export default MDXContent
...
+[`micromark-extension-mdxjs`](https://github.com/micromark/micromark-extension-mdxjs)
+(which in turn uses [acorn][]).
+Then [`mdast-util-from-markdown`](https://github.com/syntax-tree/mdast-util-from-markdown)
+and its extension
+[`mdast-util-mdx`](https://github.com/syntax-tree/mdast-util-mdx) are used to
+turn the results from the parser into a syntax tree:
+[mdast](https://github.com/syntax-tree/mdast).
+
+Markdown is closest to the source format.
+This is where remark plugins come in.
```
Comment (ChristianMurphy) [inline readme.md:1764]: it may be good to link to https://github.com/remarkjs/remark/blob/main/doc/plugins.md#plugins

File: readme.md [hunk 2]
```diff
@@ -1737,6 +1738,66 @@ export default MDXContent
...
+Another reason is that there are many things folks need when they go MDX -> JS,
+markdown -> HTML, or even folks who only process their HTML -> HTML: use cases
+other than xdm.
+By having a single AST in these cases and writing a plugin that works on that
+AST, that plugin can supports *all* these use cases (for example,
+[`rehype-highlight`](https://github.com/rehypejs/rehype-highlight)
+for syntax highlighting or
+[`rehype-katex`](https://github.com/remarkjs/remark-math/tree/main/packages/rehype-katex)
+for math).
+So, this is where rehype plugins come in: most of the plugins, probably.
```
Comment (ChristianMurphy) [inline readme.md:1786]: It may be good to link https://github.com/rehypejs/rehype/blob/main/doc/plugins.md#plugins

**Review by ChristianMurphy — APPROVED** [review]
Looks good, it could also be helpful to get a visual in here, similar to: https://github.com/remarkjs/react-remark#architecture 
http://asciiflow.com/ is a quick way to generate text diagrams

**Comment by wooorm** [comment]
@ChristianMurphy It becomes a rather wide, boring, graph:

````
```ascii
           +--------+        +----------------+        +---------------+       +----------------+       +--------------+         +-----------------+                  |
-markdown->+ remark +-mdast->+ remark plugins +-mdast->+ remark-rehype +-hast->+ rehype plugins +-hast->+ rehype-recma +-esast-> | recma-stringify |
           +--------+   ^    +----------------+   ^    +---------------+   ^   +----------------+   ^   +--------------+
                        |                         |                        |                        |
                  w/ embedded JS                  "                        "                        "
```
````

Maybe asking folks to look at `core.js` at the end is more useful? https://github.com/wooorm/xdm/blob/f43715f20fb79b4fce95eb81ee6a82a6d86df227/lib/core.js#L69-L95

**Comment by ChristianMurphy** [comment]
wide yes, boring maybe.
But I see it as being a quick reference to where the different pieces fit together.
I'd personally lean towards the wide and "boring" diagram over "go read the code", but that's just me :man_shrugging:

**Comment by wooorm** [comment]
> "boring" diagram

I mean that word more in the sense that turning `core.js` into a diagram as I tried above makes a graph that‘s full jargon but none of the interesting things. There’s already a more detailed explanation of these steps in prose, where links and notes can be added.

For the “summary” function that a diagram could have, I opted to use a [list in prose](https://github.com/wooorm/xdm/pull/6/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R1752)

**Pull request merged**